MahApps.Metro/Controls/WindowSettings.cs

```diff
@@ -64,9 +64,17 @@ public bool UpgradeSettings
         {
             get
             {
-                if (this["UpgradeSettings"] != null)
+                try
                 {
-                    return (bool)this["UpgradeSettings"];
+                    if (this["UpgradeSettings"] != null)
+                    {
+                        return (bool)this["UpgradeSettings"];
+                    }
+                }
+                catch (ConfigurationErrorsException ex)
+                {
+                    var filename = ((ConfigurationErrorsException)ex.InnerException).Filename;
+                    throw new MahAppsException(string.Format("The settings file {0} seems to be corrupted", filename), ex);
                 }
```

Comment:
Yes they are the same exception type but in the outer exception Filename is null. You really have to go to the InnerException and cast it… I just debugged the code again to be sure about that.

 

Von: Jan Karger [mailto:notifications@github.com] 
Gesendet: Freitag, 4. September 2015 21:24
An: MahApps/MahApps.Metro <MahApps.Metro@noreply.github.com>
Cc: Markus <markus.remmel@remolution-software.com>
Betreff: Re: [MahApps.Metro] If the user.config file is corrupt the whole application crashes (#2112)

 

In MahApps.Metro/Controls/WindowSettings.cs <https://github.com/MahApps/MahApps.Metro/pull/2112#discussion_r38785880> :

>                  {
> -                    return (bool)this["UpgradeSettings"];
> +                    if (this["UpgradeSettings"] != null)
> +                    {
> +                        return (bool)this["UpgradeSettings"];
> +                    }
> +                }
> +                catch (ConfigurationErrorsException ex)
> +                {
> +                    var filename = ((ConfigurationErrorsException)ex.InnerException).Filename;
> +                    throw new MahAppsException(string.Format("The settings file {0} seems to be corrupted", filename), ex);
>                  }

sorry for my nitpicking ;-) but why must the inner exception cast to ConfigurationErrorsException? the ex is already a ConfigurationErrorsException?

should this better so?

catch (ConfigurationErrorsException ex)
{
    var filename = ex.Filename;
    throw new MahAppsException(string.Format("The settings file {0} seems to be corrupted", filename), ex.InnerException);
}

—
Reply to this email directly or view it on GitHub <https://github.com/MahApps/MahApps.Metro/pull/2112/files#r38785880> .  <https://github.com/notifications/beacon/AD73r6m86DZcFr3Smn9bz9rsLANCtqnRks5ouedigaJpZM4F3__-.gif>